src/components/ui/NavItem.tsx

```diff
@@ -0,0 +1,13 @@
+interface NavItemProps {
+  iconClass: string;
+  label: string;
+}
+
+const NavItem: React.FC<NavItemProps> = ({ iconClass, label }) => (
```

Comment:
I think the usage of React.FC was one of those controversial discussions we had along of the years, but I prefer this approach while declarimg my types

```suggestion
const NavItem = ({ iconClass, label }: NavItemProps) => (
```
You can do some research about the pros/cons of using react fc, since now is not as awful as before I think you can avoid it in the future

https://www.totaltypescript.com/you-can-stop-hating-react-fc

Reply:
Yes React.FC is not as used as before. Will adapt my code to avoid its usage